ratbag_emu/hw_component.py

```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: MIT
+
+import logging
+import typing
+
+from typing import Any
+
+if typing.TYPE_CHECKING:
+    from ratbag_emu.device import Device  # pragma: no cover
+
+
+class HWComponent(object):
+    '''
+    Represents a physical hardware component
+
+    This is the "brain" of the device, it is here where we custom logic is
```

Comment:
typo and we better write this as `This is the "brain" of the device. The custom logic is implemented here`